src/blockaxis.jl

```diff
@@ -6,6 +6,8 @@
     b[block(K)][K.α...]
 @inline getindex(b::AbstractArray, K::BlockIndex{1}, J::BlockIndex{1}...) =
     b[BlockIndex(tuple(K, J...))]
+getindex(b::AbstractArray{T,N}, K::BlockIndex{1}) where {T,N} =
+    throw(DimensionMismatch("Incorrect blockindex of dimension 1 into into array of dimension $(N)"))
```

Comment:
Agreed it shouldn't `StackOverflow` but perhaps we want it to return:
```julia
julia> A[Block(1)][1]
1.173526542790686
```
Perhaps a simple (though inefficient) solution is to change this to `b[block(K)][blockindex(K)]`?

Reply:
I agree, and I thought it was going to be a simple fix, like this:
```julia
@inline getindex(b::AbstractArray{T,N}, K::BlockIndex{1}) where {T,N} =
    b[Block(K.I[1])][K.α[1]]
```

But when writing tests, there's another issue that prevents this:
```julia
julia> using BlockArrays

julia> A = [1 2; 3 4]
2×2 Array{Int64,2}:
 1  2
 3  4

julia> blocksize(A)
(1, 1)

julia> A[Block(1)]
2-element Array{Int64,1}:
 1
 3

julia> A[Block(1,1)]
2×2 Array{Int64,2}:
 1  2
 3  4
```
The problem here is that `A[Block(1)]` and `A[Block(1,1)]` don't agree, and in addition `A[Block(2)]` throws an error because of the `blocksize` of `A`, so defining `A[BlockIndex(2,1)] = A[Block(2)][1]` wouldn't work.

This is probably best discussed in a separate issue.